### PR TITLE
Fix jshint warnings

### PR DESCRIPTION
**Pull request by 5earle**
## PR Checklist
Please check that your PR fulfills the following requirements:

- [x] The commit message follows [our guidelines](https://github.com/angular/material/blob/master/.github/CONTRIBUTING.md#-commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: 9153

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by 5earle** [comment]
Addresses [#9153](https://github.com/angular/material/issues/9153) 

Remove JSHint warnings

**Comment by 5earle** [comment]
ok so just make changes to the .jshint and the missing semicolons?

**Comment by Splaktar** [comment]
Yes please, also the `Confusing use of '!'`.

File: .jshintrc [hunk 1]
```diff
@@ -1,4 +1,7 @@
 {
+  "sub": true,
+  "multistr": true,
+  "-W018": true,
```
Comment (Splaktar) [inline .jshintrc]: Sorry, I was trying to say that we should fix the issues where "confusing use of '!'" gave a warning, not exclude the rule.

File: src/components/autocomplete/demoBasicUsage/script.js [hunk 2]
```diff
@@ -55,12 +55,12 @@
...
-              Nebraska, Nevada, New Hampshire, New Jersey, New Mexico, New York, North Carolina,\
-              North Dakota, Ohio, Oklahoma, Oregon, Pennsylvania, Rhode Island, South Carolina,\
-              South Dakota, Tennessee, Texas, Utah, Vermont, Virginia, Washington, West Virginia,\
-              Wisconsin, Wyoming';
+               Florida, Georgia, Hawaii, Idaho, Illinois, Indiana, Iowa, Kansas, Kentucky, Louisiana,\
+               Maine, Maryland, Massachusetts, Michigan, Minnesota, Mississippi, Missouri, Montana,\
+               Nebraska, Nevada, New Hampshire, New Jersey, New Mexico, New York, North Carolina,\
+               North Dakota, Ohio, Oklahoma, Oregon, Pennsylvania, Rhode Island, South Carolina,\
+               South Dakota, Tennessee, Texas, Utah, Vermont, Virginia, Washington, West Virginia,\
+               Wisconsin, Wyoming';
```
Comment (Splaktar) [inline src/components/autocomplete/demoBasicUsage/script.js]: Can you please remove these added spaces? It should be indented 8 spaces, not 9.

File: .jshintrc (hunk 1 above)
Comment (5earle) [reply .jshintrc]: Ok i will look into it, but I'm not sure why its showing a warning

**Comment by 5earle** [comment]
these are giving me some problems to fix

if (!!value === targetValue)
if (setVisible.queued && setVisible.value === !!value

**Comment by Splaktar** [comment]
OK, yeah it's not trivial to fix those. Let's go with your first instinct and just exclude that rule in the `.jshintrc` file for now. Thanks for looking into it!

**Comment by Splaktar** [comment]
I would like to get this merged into 1.1.8 as we're hoping to release it this week, can you please squash your commits so that we can merge this?

**Comment by 5earle** [comment]
ok so i was trying to do the rebase.

after squash attempt:

commit 87526948513947c41c5b92c84f643ecf790abac8
Author: ew <dancehallvid@gmail.com>
Date:   Mon Feb 5 22:43:56 2018 -0500
    Fix jshint warnings
    add sub:true and multistr=true to .jshintrc
    revert and remove confusing use of ! warning
    update fix
    fix space
    exclude warnings
***
Fail to Push:

 ! [rejected]            fix-jshint -> fix-jshint (non-fast-forward)
error: failed to push some refs to 'https://github.com/5earle/material.git'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.

**Comment by Splaktar** [comment]
Have you tried force push? That's needed when re-writing a remote branch's history like we want to do when squashing commits.

**Comment by 5earle** [comment]
ok that did the trick

**Pull request merged**